lib/ui/window.dart

```diff
@@ -210,7 +210,13 @@ class Window {
   /// [physicalSize], or [padding] values change, for example when the device is
   /// rotated or when the application is resized (e.g. when showing applications
   /// side-by-side on Android).
-  VoidCallback onMetricsChanged;
+  VoidCallback get onMetricsChanged => _onMetricsChanged;
+
+  set onMetricsChanged(VoidCallback callback) {
+    _onMetricsChanged = _zonedVoidCallback(callback);
+  }
+
+  VoidCallback _onMetricsChanged;
```

Comment:
For each setter you've added, can you add some dartdocs to the effect of (maybe phrase it better, this is not well said):
```
/// Replaces the callback. The callback will be invoked in the same [Zone] as the setter's caller.
```